src/public/js/util.js

```diff
@@ -185,13 +256,89 @@ var UTIL = (function() {
         throw new Error('Division did not happen properly');
       }
     }
+
     return remainder;
   }
 
+  // Returns evaluation in GF(256) of polynomial at value
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialEval(polynomial, value) {
+    var valuePower = 1; // updated to represent value^i during loop
+    var result = 0;
+    var term;
+    var coefficient;
+    for (var i = 0; i < polynomial.length; i++) {
```

Comment:
I personally prefer starting at 1 and going to <= length for this type of loop. Theoretically it may be a tiny bit more efficient, but I also just like to minimize the number of plus and minus constants in array indexes.
Up to you on how you want that to look.